apps/server/src/modules/tool/controller/dto/request/external-tool-post.params.ts

```diff
@@ -1,15 +1,15 @@
 import { IsArray, IsBoolean, IsOptional, IsString, ValidateNested } from 'class-validator';
 import { ApiExtraModels, ApiProperty, ApiPropertyOptional, getSchemaPath } from '@nestjs/swagger';
 import { Type } from 'class-transformer';
-import { CustomParameterCreateParams } from './custom-parameter.params';
+import { CustomParameterPostParams } from './custom-parameter.params';
 import { BasicToolConfigParams } from './basic-tool-config.params';
 import { ExternalToolConfigCreateParams } from './external-tool-config.params';
 import { Oauth2ToolConfigParams } from './oauth2-tool-config.params';
 import { ToolConfigType } from '../../../interface';
 import { Lti11ToolConfigParams } from './lti11-tool-config.params';
 
 @ApiExtraModels(Lti11ToolConfigParams, Oauth2ToolConfigParams, BasicToolConfigParams)
-export class ExternalToolParams {
+export class ExternalToolPostParams {
```

Comment:
why are name, isHidden, and openNewTab required for update requests? wouldnt it make more sense to require these only for create, but have all properties optional for updates?

You can achieve this easily by using Partial<T> type, which defines a type that is the same as T, but with all properties optional.

Reply:
Yes, in case of our client we have all props. But for other calls your are right. I defined a new type